src/rgw/rgw_bucket_layout.h

```diff
@@ -271,16 +277,28 @@ inline uint32_t num_shards(const bucket_index_normal_layout& index) {
   // old buckets used num_shards=0 to mean 1
   return index.num_shards > 0 ? index.num_shards : 1;
 }
+inline uint32_t min_layout_shards(const bucket_index_normal_layout& index) {
+  // old buckets used num_shards=0 to mean 1
+  return index.min_num_shards > 0 ? index.min_num_shards : 1;
```

Comment:
we won't have to deal with old buckets that set min_num_shards=0, so this can probably just return `index.min_num_shards` directly